compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -9758,7 +9758,9 @@ private void handleSafeNavigation(BLangBlockStmt blockStmt, BLangAccessExpressio
             if (this.safeNavigationAssignment != null) {
                 this.safeNavigationAssignment.expr = addConversionExprIfRequired(accessExpr, tempResultVar.getBType());
             }
-            return;
+            if (Types.getReferredType(originalType).tag != TypeTags.JSON) {
```

Comment:
@lasinicl, Is this the only type that we need to disallow here?

Also, I have another concern. Please consider the below example (`json` is replaced by `map<any>`). 
```ballerina
import ballerina/io;

type FooRec1 record {
    map<any> j;
};

type FooRec3 record {
    json j?;
    FooRec1 fooRec?;
};

public function main() returns error? {
    FooRec3 rec2 = {j: "1", fooRec: {j: {k: "3", l: {m: "4"}}}};
    any anyResult = rec2.fooRec?.j["k"];
    io:println(anyResult);
}
```
For this example, the `if` statement in L:9751 passes, because `accessExpr.nilSafeNavigation` is true. Can you check whether we have set the proper values for `errorSafeNavigation` and `nilSafeNavigation`?

Reply:
This may be correct since it is kinda like

```ballerina
map<any>? j = {};
any anyResult = j["k"];
```

And we don't allow
```ballerina
import ballerina/io;

type FooRec1 record {
    record {|
        int k;
    |} j;
};

type FooRec3 record {
    json j?;
    FooRec1 fooRec?;
};

public function main() returns error? {
    FooRec3 rec2 = {j: "1", fooRec: {j: {k: 1}}};
    any anyResult = rec2.fooRec?.j.k;
}
```